ci/build_cpp.sh

```diff
@@ -16,8 +16,28 @@ rapids-logger "Begin cpp build"
 
 sccache --zero-stats
 
-# This calls mambabuild when boa is installed (as is the case in the CI images)
-RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rapids-conda-retry mambabuild conda/recipes/librmm
+RAPIDS_PACKAGE_VERSION=$(rapids-generate-version)
+export RAPIDS_PACKAGE_VERSION
+
+# rattler-build doesn't have built-in support for GIT_DESCRIBE_NUMBER and
+# GIT_DESCRIBE_HASH so we set them in the environment first
+# shellcheck disable=SC2034
+IFS=- read -r TAG_VERSION GIT_DESCRIBE_NUMBER GIT_DESCRIBE_HASH <<< "$(git describe --tags)"
+unset TAG_VERSION
+export GIT_DESCRIBE_NUMBER
+export GIT_DESCRIBE_HASH
+
+mamba install rattler-build -c conda-forge -y
+
+rattler-build build --recipe conda/recipes/librmm \
+                    --experimental \
+                    --no-build-id \
+                    --channel-priority disabled \
+                    --output-dir "$RAPIDS_CONDA_BLD_OUTPUT_DIR"
+                    # These are probably set via `rapids-configure-conda-channels`
+                    # -c rapidsai \
+                    # -c conda-forge \
+                    # -c nvidia
```

Comment:
These _are_ set via `rapids-configure-conda-channels` BUT, for the python builds where we also specify `${CPP_CHANNEL}` that includes an implicit `--override-channels`, so we need to list them explicitly